syntax = "proto3";

package cart;
import "common.proto";

service CartService {
  rpc GetCart(CartRequest) returns (CartResponse);
  rpc AddItem(AddItemRequest) returns (CartResponse);
  rpc RemoveItem(RemoveItemRequest) returns (CartResponse);
  rpc UpdateItem(UpdateItemRequest) returns (CartResponse);
  rpc ClearCart(CartRequest) returns (common.StatusResponse);
}

message CartRequest {
  string user_id = 1;
}

message AddItemRequest {
  string user_id = 1;
  string product_id = 2;
  int32 quantity = 3;
}

message RemoveItemRequest {
  string user_id = 1;
  string product_id = 2;
}

message UpdateItemRequest {
  string user_id = 1;
  string product_id = 2;
  int32 quantity = 3;
}

message CartItem {
  string product_id = 1;
  string name = 2;
  double price = 3;
  int32 quantity = 4;
  string image_url = 5;
}

message CartResponse {
  string user_id = 1;
  repeated CartItem items = 2;
  double total_amount = 3;
  int32 total_items = 4;
}
